doc/Container Hardening/SONiC_container_hardening_HLD.md

```diff
@@ -0,0 +1,349 @@
+
+In the docker-database.mk file adjust this line:
+
+        $(DOCKER_DATABASE)_RUN_OPT += -t –-cap-add NET_ADMIN  # Changed by removing the --privileged flag and adding --cap-add flag
+
+### 4.2 net=host optimization
+
+Here we will provide a detailed example of how to switch from the `--net=host` configuration (host network) to the `--net=bridge` configuration paired with port forwarding in a specific container. We are using the database container as an example for this item.
+
+The original docker creation should be like in the example below:
+
+docker with host sharing:
+
+	docker create --privileged -t -v /etc/sonic:/etc/sonic:ro  \
+					--net=$NET \
+					-e RUNTIME_OWNER=local \
+					--uts=host \
+					--log-opt max-size=2M --log-opt max-file=5 \
+					--tmpfs /tmp \
+					$DB_OPT \
+					$REDIS_MNT \
+					-v /usr/share/sonic/device/$PLATFORM:/usr/share/sonic/platform:ro \
+					--tmpfs /var/tmp \
+					--env "NAMESPACE_ID"="$DEV" \
+					--env "NAMESPACE_PREFIX"="$NAMESPACE_PREFIX" \
+					--env "NAMESPACE_COUNT"=$NUM_ASIC \
+					--name=database_no_net \
+					--cap-drop=NET_ADMIN \
+					docker-database:latest
+
+To disable the sharing of the networking stack between the host and a container we need to remove the flag: `--net=host`. Because we have not specified any `--network` flag, the containers connect to the default bridge network `--net=bridge`.
+To support port forwarding we are required to add the flag:  `-p <port>:<port>`
+
+The "new" docker creation file database.sh can be seen in the code block below:
+
+Docker with port forwarding and default bridge network
+
+	docker create --privileged -t -v /etc/sonic:/etc/sonic:ro  \
+					**-p 6379:6379** \
+					-e RUNTIME_OWNER=local \
+					--uts=host \
+					--log-opt max-size=2M --log-opt max-file=5 \
+					--tmpfs /tmp \
+					$DB_OPT \
+					$REDIS_MNT \
+					-v /usr/share/sonic/device/$PLATFORM:/usr/share/sonic/platform:ro \
+					--tmpfs /var/tmp \
+					--env "NAMESPACE_ID"="$DEV" \
+					--env "NAMESPACE_PREFIX"="$NAMESPACE_PREFIX" \
+					--env "NAMESPACE_COUNT"=$NUM_ASIC \
+					--name=$DOCKERNAME \
+					docker-database:latest \
+
+**How we did it?**
+
+To create a docker with the flags above it is required to set the "new" flag in the file docker_image_ctl.js. Follow the call `docker create {{docker_image_run_opt}} \`: 
+and replace the `–--net=$NET`.
+docker flag generation
+
+	{%- if docker_container_name != "database" %}
+					--net=$NET \
+	{%- endif %}
+	{%- if docker_container_name == "database" %}
+					-p 6379:6379 \
+	{%- endif %}
+
+#### How to check?
+
+Go into the docker - `docker exec -it docker bash`
+Run `ifconfig`.
+
+On a docker with host network - you'll be able to view all physical interfaces.
+On a docker without host network - we'll see only eth0 and lo.
+
```

Comment:
I wonder whether syslog(especially send syslog package to remote rsyslog server) can work well with this change?

Reply:
Syslog can function with this.
Also, the syslog is currently running on top of the host - where nothing is changed there.
So I'm going to split my answer:
1. There should be no impact on rsyslog and none of these items affect it as it runs on top of the host
2. In the same manner that we changed SNMP - we can change rsyslog (if there is a decision to move it to a container based solution)

Reply:
It should be clarified the syslog I mean is generated inside container. I just tried image built by this snmp network mode related PR: https://github.com/sonic-net/sonic-buildimage/pull/15176 (SONiC.master-15176.360116-6bb24c135). And cannot find any syslog generated by snmp container (contains 'snmp#') in local host syslog file or our syslog server. Not sure whether you have verified this, could you confirm that?

Reply:
Can you please refer this item to the specific PR?
If there is some implementation concern that you have noticed - I'd appreciate it if you could direct it to the relevant PR.
This is a good item and I do not recall if this was reviewed thoroughly (might have missed something)